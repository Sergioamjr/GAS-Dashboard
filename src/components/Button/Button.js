import React from "react";

const mountClassNames = props => {
  const {
    className = "",
    small,
    icon,
    type,
    circle,
    noborder,
    ...otherProps
  } = props;

  let btnClassName = `btn ${className}`;

  type && (btnClassName = `btn-${type} ${btnClassName}`);
  icon && (btnClassName = `btn-icon ${btnClassName}`);
  small && (btnClassName = `btn-small ${btnClassName}`);
  circle && (btnClassName = `btn-circle ${btnClassName}`);
  noborder && (btnClassName = `btn-noborder ${btnClassName}`);

  return { className: btnClassName, otherProps };
};

const Button = props => {
  const { className, otherProps } = mountClassNames(props);
  const { children, disabled, loading } = otherProps;
  return (
    <button
      {...otherProps}
      disabled={disabled || loading}
      className={className}
    >
      {children}
      {!!loading && <i className="m-left-10 fas fa-sync rotate" />}
    </button>
  );
};

export default Button;
