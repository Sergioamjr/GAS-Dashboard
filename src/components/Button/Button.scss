.btn {
  padding: 10px 20px;
  border: 0;
  border-radius: 4px;
  font-size: 14px;
  cursor: pointer;
  color: $dark;
  border: 1px solid $lightDarken;
  &:hover {
    background: $lightDarken;
  }
  &-primary {
    background: $primary;
    color: $white;
    &:hover {
      background: darken($primary, 10%);
    }
  }
  &-danger {
    background: $danger;
    color: $white;
    &:hover {
      background: darken($danger, 10%);
    }
  }

  &-success {
    background: $success;
    color: $white;
    &:hover {
      background: darken($success, 10%);
    }
  }
  &-small {
    padding: 5px 10px;
  }
  &-icon {
    padding: 10px;
  }

  &-circle {
    border-radius: 50%;
    padding: 0;
    width: 30px;
    height: 30px;
  }
  &-noborder {
    border-radius: 0;
  }
}
