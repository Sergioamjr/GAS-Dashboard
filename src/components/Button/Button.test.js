import React from "react";
import { shallow } from "enzyme";
import Button from "./Button";
import renderer from "react-test-renderer";

describe("<Button />", () => {
  it("should create a snapshot file", () => {
    const button = renderer.create(<Button>Botão Padrão</Button>).toJSON();
    const smallButton = renderer
      .create(<Button small>Botão pequeno</Button>)
      .toJSON();
    const primaryButton = renderer
      .create(<Button type="primary">Botão primário</Button>)
      .toJSON();

    expect(button).toMatchSnapshot();
    expect(smallButton).toMatchSnapshot();
    expect(primaryButton).toMatchSnapshot();
  });

  it("should render Button component", () => {
    shallow(<Button />);
  });

  it("should render a small button", () => {
    const wrapper = shallow(<Button small />);
    expect(wrapper).toHaveLength(1);
    expect(wrapper.hasClass("btn-small")).toBeTruthy();
  });

  it("should render a button of type primary", () => {
    const wrapper = shallow(<Button type="primary">Primário</Button>);
    expect(wrapper).toHaveLength(1);
    expect(wrapper.hasClass("btn-primary")).toBeTruthy();
  });
});
