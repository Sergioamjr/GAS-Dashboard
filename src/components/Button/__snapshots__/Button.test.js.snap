// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should create a snapshot file 1`] = `
<button
  className="btn "
>
  Botão Padrão
</button>
`;

exports[`<Button /> should create a snapshot file 2`] = `
<button
  className="btn-small btn "
>
  Botão pequeno
</button>
`;

exports[`<Button /> should create a snapshot file 3`] = `
<button
  className="btn-primary btn "
>
  Botão primário
</button>
`;
